File: samples/three-layer-kmp-domain-data/iosApp/Sources/HomeViewModel.swift
import SharedDomain

final class HomeViewModel {
    private let useCase = GetTasksUseCase()

    func screenState(platform: String) -> String {
        let task = useCase.execute(platform: platform)
        return "Native HomeViewModel -> \(task.title)\n\(task.details)"
    }

    func refresh(platform: String) -> String {
        let task = useCase.execute(platform: platform)
        return "Native HomeViewModel refreshed -> \(task.title)\n\(task.details)"
    }
}

File: samples/18_three-layer-kmp-domain-data/desktopApp/src/main/kotlin/com/example/kmpsamples/threelayerkmpdomaindata/desktop/HomeViewModel.kt
package com.example.kmpsamples.threelayerkmpdomaindata.desktop

import com.example.kmpsamples.threelayerkmpdomaindata.shareddomain.GetTasksUseCase

class HomeViewModel(
    private val useCase: GetTasksUseCase = GetTasksUseCase()
) {
    fun screenState(platform: String): String {
        val task = useCase.execute(platform)
        return "Native HomeViewModel -> "+ task.title + "\n" + task.details
    }

    fun refresh(platform: String): String {
        val task = useCase.execute(platform)
        return "Native HomeViewModel refreshed -> "+ task.title + "\n" + task.details
    }
}
